beyond-all-reason/Beyond-All-Reason
# Target over-penetration for heavy railguns

### Work done

Gadget added that uses a repeatedly spawning projectile to simulate over-penetration of relatively soft targets. Respawned projectiles deal less damage with more gradual loss of impulse. It has some limited customization options for now.

The first unit candidate is the legsrail, which is part of this PR as the unit demo.

#### Test steps

Do a fightertest with legsrail and corak, set a high group size, around 25 to see results better. It takes about 2.5 Grunts to survive a shot from an Arquebus, so you should see some survivors get tossed a short distance by the higher arrest impulse. The legsrail should overpen targets up to around the Sumo semi-consistently, also.
